thavalon-webapp/src/components/gameCreation.tsx

```diff
@@ -0,0 +1,326 @@
+ * @param props - Required reqct props object.
+ */
+export function CreateJoinGameModal(props: CreateJoinGameProps): JSX.Element {
+    // State to keep track of whether or not the form should display.
+    // Needed to trigger closing animations when the form is leaving the modal.
+    const [showForm, setShowForm] = useState(false);
+    // State to track the overall modal state, including which form should be displayed.
+    const [modalState, setModalState] = useState(CreateJoinState.CreateJoinButtons);
+
+    /**
+     * Function handler for the success event from the child modal.
+     * Will reset state for future callers and close the modal.
+     */
+    function onSuccess(): void {
+        setModalState(CreateJoinState.CreateJoinButtons);
+        props.setOpen(false);
+    }
+
+    return (
+        <ReactModal
+            isOpen={props.show}
+            onRequestClose={() => props.setOpen(false)}
+            contentLabel="Create/Join Game Modal"
+            className="Modal"
+            overlayClassName="Overlay">
+            <div className="modalContainer">
+                <h2 className="modalHeader">Play Thavalon</h2>
+                <hr />
+                {modalState === CreateJoinState.CreateJoinButtons && CreateJoinButtons(setShowForm, setModalState)}
+                <CSSTransition
+                    in={showForm}
+                    timeout={300}
+                    classNames="create-join-form"
+                    onExited={() => setModalState(CreateJoinState.CreateJoinButtons)}
+                    unmountOnExit>
+                    <>
+                        {modalState === CreateJoinState.CreateGame && <CreateGameForm showForm={setShowForm} onSuccess={onSuccess} />}
+                        {modalState === CreateJoinState.JoinGame && <JoinGameForm showForm={setShowForm} onSuccess={onSuccess} />}
+                    </>
+                </CSSTransition>
+            </div>
+        </ReactModal>
+    );
+}
+
+/**
+ * 
+ * @param showForm Sets whether or not a form should display
+ * @param setModalState Sets the modal state to the correct form that should display
+ */
+function CreateJoinButtons(showForm: Dispatch<SetStateAction<boolean>>, setModalState: Dispatch<SetStateAction<CreateJoinState>>): JSX.Element {
+    return (
+        <div className="create-join-buttons">
+            <Container>
+                <div className="center">
+                    <ButtonGroup vertical>
+                        <Button
+                            className="with-bottom-margin"
+                            variant="primary"
+                            onClick={() => {
+                                showForm(true);
+                                setModalState(CreateJoinState.JoinGame);
+                            }}>
+                            Join Game
+                        </Button>
+                        <Button
+                            variant="primary"
```

Comment:
two primary buttons intended?

Reply:
Yes, the primary is the styling. In this case, it's two blue buttons. If you want other buttons, I can change.